File: dl-invites/app/src/main/java/com/google/firebase/dynamicinvites/presenter/InvitePresenter.java
package com.google.firebase.dynamicinvites.presenter;

import android.content.Context;
import androidx.annotation.DrawableRes;

import com.google.firebase.dynamicinvites.kotlin.model.InviteContent;

// [START ddl_invite_presenter]

/**
 * Presents the invite using a specific method, such as email or social.
 */
public class InvitePresenter {

    /**
     * The user-visible name of the invite method, like 'Email' or 'SMS'
     **/
    public final String name;

    /**
     * An icon representing the invite method.
     **/
    @DrawableRes
    public final int icon;

    /**
     * Whether or not the method is available on this device. For example, SMS is phone only.
     **/
    public final boolean isAvailable;

    /**
     * The Content of the invitation
     **/
    public final InviteContent content;

    public InvitePresenter(String name, @DrawableRes int icon, boolean isAvailable, InviteContent content) {
        // [START_EXCLUDE]
        this.name = name;
        this.icon = icon;
        this.isAvailable = isAvailable;
        this.content = content;
        // [END_EXCLUDE]
    }

    /**
     * Send the invitation using the specified method.
     */
    public void sendInvite(Context context) {
        // ...
    }

}
// [END ddl_invite_presenter]

File: dl-invites/app/src/main/java/com/google/firebase/dynamicinvites/kotlin/presenter/InvitePresenter.kt
package com.google.firebase.dynamicinvites.kotlin.presenter

import android.content.Context
import androidx.annotation.DrawableRes

import com.google.firebase.dynamicinvites.kotlin.model.InviteContent

// [START ddl_invite_presenter]
/**
 * Presents the invite using a specific method, such as email or social.
 */
open class InvitePresenter(
    /** The user-visible name of the invite method, like 'Email' or 'SMS'  */
    val name: String,
    /** An icon representing the invite method.  */
    @param:DrawableRes @field:DrawableRes
    val icon: Int,
    /** Whether or not the method is available on this device. For example, SMS is phone only.  */
    val isAvailable: Boolean,
    /** The Content of the invitation  */
    val content: InviteContent
) {
    /**
     * Send the invitation using the specified method.
     */
    open fun sendInvite(context: Context) {
        // ...
    }
}
// [END ddl_invite_presenter]
